### PR TITLE
add new method to HabitAssignController:

### PR DESCRIPTION
**Pull request by Skajl-dev**
- method to find all user activity between 2 dates
(@GetMapping("/activity/{from}/to/{to}"))

### CODE REVIEW

File: service/src/main/java/greencity/service/HabitAssignServiceImpl.java [hunk 1]
```diff
@@ -412,4 +424,62 @@ private int countNewHabitStreak(List<HabitStatusCalendar> habitCalendars) {
...
+                        isDateEnrolled = true;
+                    }
+                }
+                if (!isDateEnrolled) {
+                    dto.getHabitAssigns().add(HabitEnrollDto.builder()
+                        .habitDescription(habitTranslation.getDescription()).habitName(habitTranslation.getName())
+                        .isEnrolled(false).habitId(habitAssign.getHabit().getId()).build());
+                }
+            }
+        }
```
Comment (vterlyha) [inline service/src/main/java/greencity/service/HabitAssignServiceImpl.java:483]: Don't understand what is being done here. Split to several private methods.

File: service/src/main/java/greencity/service/HabitAssignServiceImpl.java [hunk 2]
```diff
@@ -412,4 +424,62 @@ private int countNewHabitStreak(List<HabitStatusCalendar> habitCalendars) {
...
             habitAssign -> buildHabitAssignDto(habitAssign, language)).collect(Collectors.toList());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<HabitsDateEnrollmentDto> findActiveHabitAssignsBetweenDates(Long userId, LocalDate from, LocalDate to,
+        String language) {
+        List<HabitAssign> list = habitAssignRepo.findAllActiveHabitAssignsBetweenDates(userId, from, to);
```
Comment (vterlyha) [inline service/src/main/java/greencity/service/HabitAssignServiceImpl.java:434]: list - is a bad name

File: service/src/main/java/greencity/service/HabitAssignServiceImpl.java (hunk 1 above)
Comment (Skajl-dev) [reply service/src/main/java/greencity/service/HabitAssignServiceImpl.java]: Thanks, changed

File: service/src/main/java/greencity/service/HabitAssignServiceImpl.java (hunk 2 above)
Comment (Skajl-dev) [reply service/src/main/java/greencity/service/HabitAssignServiceImpl.java]: Thanks, changed

**Review by Yura09 — APPROVED** [review]

**Review by forestik — APPROVED** [review]

**Review by airetOK — APPROVED** [review]

**Pull request merged**